### PR TITLE
Change Pub/Sub to Action Cable

### PR DESCRIPTION
**Pull request by OmegaLord**
Change Pub/Sub to Action Cable
Change .js to ES 6

### CODE REVIEW

File: app/javascript/packs/answers.js [hunk 1]
```diff
@@ -1,144 +1,159 @@
...
-    answer.comments.forEach(function (comment) {
-      $(block).find('.comments').append(comment_block(comment))
-    })
-    if (typeof (gon.user_id) !== 'undefined') {
-      new_comment = new_comment_form.content.cloneNode(true)
-      $(new_comment).find('#new_comment_answer')
-        .attr('action', '/questions/' + answer.question_id + '/answers/' + answer.id + '/comments')
-      $(block).find('.answer').append($(new_comment))
+    function updateAnswer (answer) {
+      const prevElem = $('*[data-answer-id="' + answer.id + '"]').prev()
```
Comment (AZatsepa) [inline app/javascript/packs/answers.js]: `*[data-answer-id="${answer.id}"]`
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/template_strings

File: app/javascript/packs/answers.js [hunk 2]
```diff
@@ -1,144 +1,159 @@
...
+      $('*[data-answer-id="' + answer.id + '"]').remove()
```
Comment (AZatsepa) [inline app/javascript/packs/answers.js]: The same

File: app/javascript/packs/answers.js [hunk 3]
```diff
@@ -1,144 +1,159 @@
...
     }
-  }
 
-  function answer_block (answer, block) {
-    var link = $(block).find('.edit-answer-link')
-    if (gon.user_id === answer.user_id) {
-      var link = $(block).find('.edit-answer-link')
-      link.next().attr('href', '/questions/' + answer.question_id + '/answers/' + answer.id)
+    function destroyAnswer (answer) {
+      $('*[data-answer-id="' + answer.id + '"]').remove()
```
Comment (AZatsepa) [inline app/javascript/packs/answers.js]: The same

File: app/javascript/packs/questions.js [hunk 4]
```diff
@@ -1,70 +1,77 @@
...
+        $(e.currentTarget).closest('form').children('.files').append(divClone)
+      }
+    })
 
-  function incrementString (str) {
-    return str.replace(/\d/, function (s) {
-      return parseInt(s) + 1
+    App.cable.subscriptions.create('QuestionsChannel', {
+      received (data) {
+        const action = data.action
```
Comment (AZatsepa) [inline app/javascript/packs/questions.js]: https://developer.mozilla.org/uk/docs/Web/JavaScript/Reference/Operators/%D0%94%D0%B5%D1%81%D1%82%D1%80%D1%83%D0%BA%D1%82%D1%83%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D1%96%D1%8F

File: config/environments/production.rb [hunk 5]
```diff
@@ -91,4 +91,9 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+
+  config.action_cable.mount_path = '/cable'
+  config.action_cable.url = 'wss://localhost:3000/cable'
+  config.action_cable.allowed_request_origins = [ /http:\/\/localhost:*/ ]
```
Comment (AZatsepa) [inline config/environments/production.rb]: Not sure that it will work on production

Comment (OmegaLord) [reply config/environments/production.rb]: To not forget this settings, I will change them when deploy

**Pull request merged**